lms/djangoapps/discussion/rest_api/serializers.py

```diff
@@ -475,48 +478,65 @@ class DiscussionSettingsSerializer(serializers.Serializer):
     always_divide_inline_discussions = serializers.BooleanField()
     division_scheme = serializers.CharField()
 
-    def __init__(self, *args, **kwargs):
-        self.course = kwargs.pop('course')
-        self.discussion_settings = kwargs.pop('discussion_settings')
-        super().__init__(*args, **kwargs)
-
-    def validate(self, attrs):
+    def to_internal_value(self, data: dict) -> dict:
         """
-        Validate the fields in combination.
+        Transform the *incoming* primitive data into a native value.
         """
-        if not any(field in attrs for field in self.fields):
+        payload = super().to_internal_value(data) or {}
+        if not any(field in data for field in self.fields):
             raise ValidationError('Bad request')
-
-        settings_to_change = {}
-        divided_course_wide_discussions, divided_inline_discussions = get_divided_discussions(
-            self.course, self.discussion_settings
-        )
-
-        if any(item in attrs for item in ('divided_course_wide_discussions', 'divided_inline_discussions')):
-            divided_course_wide_discussions = attrs.get(
+        course = self.context['course']
+        instance = self.context['settings']
+        if any(item in data for item in ('divided_course_wide_discussions', 'divided_inline_discussions')):
+            divided_course_wide_discussions, divided_inline_discussions = get_divided_discussions(
+                course, instance
+            )
+            divided_course_wide_discussions = data.get(
                 'divided_course_wide_discussions',
                 divided_course_wide_discussions
             )
-            divided_inline_discussions = attrs.get('divided_inline_discussions', divided_inline_discussions)
-            settings_to_change['divided_discussions'] = divided_course_wide_discussions + divided_inline_discussions
-
+            divided_inline_discussions = data.get('divided_inline_discussions', divided_inline_discussions)
+            payload['divided_discussions'] = divided_course_wide_discussions + divided_inline_discussions
         for item in ('always_divide_inline_discussions', 'division_scheme'):
-            if item in attrs:
-                settings_to_change[item] = attrs[item]
-        attrs['settings_to_change'] = settings_to_change
```

Comment:
We no longer rely on this side-effect;
we just use `to_internal_representation` as intended.